### PR TITLE
If size / offset are out of bounds just do a plain count

### PR DESCRIPTION
**Pull request by martijnvg**
so that we at least include total hits, otherwise also total hits would be equal to zero.

PR for #20501

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java [hunk 1]
```diff
@@ -110,6 +108,10 @@ public InnerHitsContext innerHits() {
         public void setChildInnerHits(Map<String, InnerHitsContext.BaseInnerHits> childInnerHits) {
             this.childInnerHits = new InnerHitsContext(childInnerHits);
         }
+
+        boolean doCountOnly() {
+            return from() < 0 || from() >= size() || size() <= 0;
+        }
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java]: This does not make sense to me: fi there are 7 inner hits and from=2 and size=1, then we should not just count?

**Review by jpountz — CHANGES_REQUESTED** [review]
This does not seem to fix the right bug?

File: core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java (hunk 1 above)
Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java]: doh... Yes, you're absolutely right.

Comment (martijnvg) [inline core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java]: @jpountz The reason I took this approach, was that `TopDocsCollector#topDocs(...)` returns a top docs with total hits set to 0 if `from` is larger than number of collected docs, which is unexpected. Instead it should just return a TopDocs with no docs but a real totalHits/maxScore?

In the search api we don't rely on `TopDocsCollector#topDocs(...)`, but rather on `TopDocs#merge`, which always set maxScore and totalHits.

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java]: > Instead it should just return a TopDocs with no docs but a real totalHits/maxScore?

I tend to agree.

**Comment by clintongormley** [comment]
@martijnvg just a reminder about this one

**Comment by dakrone** [comment]
@martijnvg are you still working on this?

**Comment by martijnvg** [comment]
This PR should have been closed as this fix is in the wrong place.

**Pull request closed without merging**